aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -935,6 +951,81 @@ std::tuple<Tensor&, Tensor&> symeig_out(Tensor& vals, Tensor& vecs, const Tensor
   return std::tuple<Tensor&, Tensor&>(vals, vecs);
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ eig ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+template <typename scalar_t>
+static void apply_eig(Tensor& vals_, Tensor& vecs_, const Tensor& self, bool eigenvectors) {
+#ifndef USE_LAPACK
+  AT_ERROR("symeig: LAPACK library not found in compilation");
+#else
+  int n = self.size(0);
+  Tensor wi = at::empty({n}, self.dtype());
+  Tensor wr = at::empty({n}, self.dtype());
+  char jobvr = eigenvectors ? 'V' : 'N';
+  int lda = n;
+  int info;
+  if (n > 0) {
+    scalar_t wkopt;
+    scalar_t* vecs_data = eigenvectors ? vecs_.data<scalar_t>() : nullptr;
+    int ldvr = eigenvectors ? n : 1;
+
+    lapackEig<scalar_t>('N', jobvr, n, self.data<scalar_t>(), lda, wr.data<scalar_t>(), wi.data<scalar_t>(),
+      nullptr, 1, vecs_data, ldvr, &wkopt, -1, &info);
+
+    int lwork = (int)wkopt;
+    Tensor work = at::empty({lwork}, self.dtype());
+
+    lapackEig<scalar_t>('N', jobvr, n, self.data<scalar_t>(), lda, wr.data<scalar_t>(), wi.data<scalar_t>(),
+      nullptr, 1, vecs_data, ldvr, work.data<scalar_t>(), lwork, &info);
+  }
+
+  Tensor wr_ = at::slice(vals_, 1, 0, 1).squeeze(1);
+  Tensor wi_ = at::slice(vals_, 1, 1, 2).squeeze(1);
+  wr_.copy_(wr);
+  wi_.copy_(wi);
+#endif
+}
+
+static std::tuple<Tensor, Tensor> _eig_helper_cpu(const Tensor& self, bool eigenvectors) {
+  TORCH_CHECK(self.dim() == 2, "A should be 2 dimensional");
+  TORCH_CHECK(self.size(0) == self.size(1), "A should be square");
+  TORCH_CHECK(self.isfinite().all().item<bool>(), "A should not contain infs or NaNs");
+  int n = self.size(0);
+  Tensor vecs_;
+  if (eigenvectors) {
+    vecs_ = at::empty({n, n}, self.dtype()).transpose(0, 1);
+  } else {
+    vecs_ = at::empty({0}, self.dtype());
+  }
+  Tensor vals_ = at::empty({n, 2}, self.dtype());
+  Tensor self_ = cloneBatchedColumnMajor(self);
+
+  AT_DISPATCH_FLOATING_TYPES(self.scalar_type(), "eig_cpu", [&]{
+    apply_eig<scalar_t>(vals_, vecs_, self_, eigenvectors);
+  });
+  return std::tuple<Tensor&, Tensor&>(vals_, vecs_);
+}
+
+std::tuple<Tensor&, Tensor&> eig_out_cpu(Tensor& vals, Tensor& vecs, const Tensor& self, bool eigenvectors) {
+  TORCH_CHECK(self.dtype() == vals.dtype(), "dtype of eigenvalues tensor must match input");
+  TORCH_CHECK(!eigenvectors || (self.dtype() == vecs.dtype()), "dtype of eigenvectors tensor must match input");
+  Tensor vals_tmp, vecs_tmp;
+  std::tie(vals_tmp, vecs_tmp) = _eig_helper_cpu(self, eigenvectors);
+  vals.resize_as_(vals_tmp).copy_(vals_tmp);
+  if (eigenvectors) {
+    // Need vector output to be in transposed-contiguous format
```

Comment:
```
if (vecs.size() != vecs_tmp.size()) {
   //to generate a warning if size is not 0
   // transpose on vecs_tmp is probably not needed,it is always square?
    at::native::resize_output(vecs, vecs_tmp.transpose(0, 1).size(), MemoryFormat::Contiguous).transpose_(0, 1);
}
vecs.copy_(vecs_tmp);
```

Reply:
The only reason I was trying to preserve the old behavior of always returning a transposed-contiguous tensor was because I thought I should avoid BC breaking changes in a porting PR. Would you consider this a BC breaking change? If not, or if this behavior change is acceptable, then I think we can just avoid the transposed-contiguous reformat altogether. In that case, we can do this:
```
at::native::resize_output(vecs, vecs_tmp.sizes());
vecs.copy_(vecs_tmp);
```

Reply:
Output strides are not part of BC guarantees (though yes, technically it's a BC breaking change). I still think when you are resizing it makes sense to make the strides of your resized output match the strides of output that lapack returns, copies of the tensors with the same layout are faster.

Reply:
Ok I see. I think my most recent commit is sufficient then.